build-tools/packages/build-tools/src/type-test-generator/main.ts

```diff
@@ -12,12 +12,100 @@ import { PackageJson } from "../common/npmPackage";
 import { buildTestCase, TestCaseTypeData } from "../typeValidator/testGeneration";
 import { getFullTypeName, getNodeTypeData, TypeData } from "../typeValidator/typeData";
 import { typeOnly } from "./compatibility";
+import { ExtractorConfig } from "@microsoft/api-extractor";
 
 // Do not check that file exists before opening:
 // Doing so is a time of use vs time of check issue so opening the file could fail anyway.
 // Do not catch error from opening file since the default behavior is fine (exits process with error showing useful message)
 const packageObject: PackageJson = readJsonSync("package.json");
+const previousPackageName = `${packageObject.name}-previous`;
+
+const previousBasePath = path.join("node_modules", previousPackageName);
+
+const previousPackageJsonPath = `${previousBasePath}/package.json`;
+
+if (!existsSync(previousPackageJsonPath)) {
+	throw new Error(
+		`${previousPackageJsonPath} not found. You may need to install the package via pnpm install. Note that type tests logic looks specifically for a package named '${previousPackageName}'`,
+	);
+}
+
+/**
+ * Retrieves the alpha trimmed file path for type definitions from the API Extractor configuration.
+ * @returns {string} The path to the alpha trimmed type definitions file.
+ * @throws {Error} If api-extractor config cannot be loaded or if the alpha trimmed file path is undefined
+ */
+function getAlphaTypeRollupPathsFromExtractorConfig(): string {
+	//Load the api-extractor
+	const extractorConfigOptions = ExtractorConfig.tryLoadForFolder({
+		startingFolder: previousBasePath,
+	});
+	if (!extractorConfigOptions || !extractorConfigOptions.configObjectFullPath) {
+		throw new Error("Failed to load extractor configuration.");
+	}
+	const apiExtractorConfigPath = extractorConfigOptions.configObjectFullPath;
+	const apiExtractorConfig = readJsonSync(extractorConfigOptions.configObjectFullPath);
+	// Resolve the api-extractor-base file path
+	const baseConfigPath = path.resolve(
+		path.dirname(apiExtractorConfigPath),
+		apiExtractorConfig.extends,
+	);
+	const baseConfig = readJsonSync(baseConfigPath);
+
+	return baseConfig.dtsRollup["alphaTrimmedFilePath"];
+}
+
+/**
+ * Extracts the type definition file path from the 'exports' field of a given package.json.
+ * Checks both 'import' and 'require' resolution methods to find the appropriate path.
+ * If the path is found, it is returned. Otherwise, an error is thrown.
+ * @param previousPackageJson
+ * @returns string - A type definition filepath based on the appropriate export.
+ */
+function getTypePathsFromExport(previousPackageJson: PackageJson): string {
```

Comment:
Nit: this is just returning a single path, so we should probably name this something like `getTypePathFromExport`.